source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-role.tsx

```diff
@@ -79,24 +79,26 @@ export function withRole<TWrappedProps extends {}>(role: ROLE, givenOptions?: IO
         }
 
         if (checkOrganizationOf) {
-          const resource = checkOrganizationOf(this.props as TWrappedProps);
+          const resource = checkOrganizationOf((this.props as any) as TWrappedProps);
 
           resultOfResource = await roleInOrganizationOfResource(currentUser, dataStore, resource, role);
         }
 
         return resultOfOrganization || resultOfResource;
       }
 
-      componentDidMount() {
-        this.doesUserHaveAccess().then(( result ) => {
+      async componentDidMount() {
```

Comment:
didn't know you could async a lifecycle function. Interesting

Reply:
yeah, the return value doesn't matter